Sources/StickeyCollectionViewLayout/StickyCollectionViewLayout.swift

```diff
@@ -0,0 +1,178 @@
+
+    private var itemAttributes = [[UICollectionViewLayoutAttributes]]()
+    private var itemsSize = [CGSize]()
+    private var contentSize : CGSize!
+
+    override public func prepare() {
+        super.prepare()
+        guard let collectionView = collectionView,
+              let numberOfSections = self.collectionView?.numberOfSections,
+              numberOfSections > 0 else {
+            return
+        }
+
+        guard itemAttributes.count > 0 else  {
+            // Calculate everything at startup. This code is only executed once
+            calculateAllItemAtrributes()
+            return
+        }
+        stickColumnHeaderAndFirstColumn()
+    }
+
+    override public var collectionViewContentSize : CGSize {
+        return self.contentSize
+    }
+    
+    override public func layoutAttributesForItem(at indexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
+        return self.itemAttributes[indexPath.section][indexPath.row]
+    }
+    
+    override public func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        var attributes = [UICollectionViewLayoutAttributes]()
+        for section in self.itemAttributes {
+            let filteredArray  = section.filter { rect.intersects($0.frame) }
+            attributes.append(contentsOf: filteredArray)
+        }
+
+        return attributes
+    }
+
+    override public func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+        return true
+    }
+    
+
+    // MARK: - Privates
+
+    private func sizeForItemWithColumnIndex(_ columnIndex: Int) -> CGSize {
+        return preferredItemSize
+    }
+    
+    private func calculateItemsSize() {
+        guard let collectionView = collectionView else {
+            return
+        }
+        for section in 0..<collectionView.numberOfSections {
+            for index in 0..<collectionView.numberOfItems(inSection: section) {
+                self.itemsSize.append(self.sizeForItemWithColumnIndex(index))
+            }
+        }
+    }
+
+    private func stickColumnHeaderAndFirstColumn() {
+        guard let collectionView = collectionView,
+            let numberOfSections = self.collectionView?.numberOfSections,
+            numberOfSections > 0 else {
+                return
+        }
+
+        let contentOffset = collectionView.contentOffset
+        for section in 0..<numberOfSections {
+            let numberOfItems : Int = collectionView.numberOfItems(inSection: section)
+            for index in 0..<numberOfItems {
+                if section != 0 && index != 0 {
+                    continue
+                }
+
+                let attributes = self.layoutAttributesForItem(at: IndexPath(item: index, section: section))!
+
+                // Stick the first row
+                if section == 0 {
+                    var frame = attributes.frame
+                    frame.origin.y = contentOffset.y
+                    attributes.frame = frame
```

Comment:
I think you can directly modify the `y` value like this `attributes.frame.origin.y = contentOffset.y`